src/Node.lua

```diff
@@ -0,0 +1,22 @@
+local Node = {}
+Node.__index = Node
+
+function Node:new()
+  return setmetatable({parent = nil, children = {}}, Node)
+end
+
+function Node:add_child(child)
+  self.children[#self.children + 1] = child
+  child.parent = self
+  child.index  = #self.children
```

Comment:
Why do you set the `index` attribute to the children's index? Is this attribute used somewhere?!